drivers/can/can_net.c

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (c) 2019 Alexander Wachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <net/can.h>
+#include <net/net_pkt.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(net_can, CONFIG_CAN_NET_LOG_LEVEL);
+
+#define CAN_NET_FILTER_NOT_SET -1
+
+struct net_can_context {
+	struct device *can_dev;
+	struct net_if *iface;
+	int rcv_filter_id;
+	int mcast_filter_id;
+};
+
+static inline u8_t get_frame_datalength(struct zcan_frame *frame)
+{
+	/* TODO: Needs update when CAN FD support is added*/
+	return frame->dlc;
+}
+
+static inline u16_t get_can_lladdr_src(struct zcan_frame *frame)
+{
+	return (frame->ext_id >> CAN_NET_IF_ADDR_SRC_POS) &
+		     CAN_NET_IF_ADDR_MASK;
+}
+
+static inline u16_t get_can_lladdr_dest(struct zcan_frame *frame)
+{
+	return (frame->ext_id >> CAN_NET_IF_ADDR_DEST_POS) &
+		     CAN_NET_IF_ADDR_MASK;
+}
+
+static inline void set_lladdr(struct net_pkt *pkt, struct zcan_frame *frame)
+{
+	struct net_buf *buf = pkt->buffer;
+
+	net_pkt_lladdr_src(pkt)->addr = buf->data;
+	net_pkt_lladdr_src(pkt)->len = sizeof(struct net_can_lladdr);
+	net_pkt_lladdr_src(pkt)->type = NET_LINK_CANBUS;
+	net_buf_add_be16(buf, get_can_lladdr_src(frame));
+	net_buf_pull(buf, sizeof(u16_t));
+
+	net_pkt_lladdr_dst(pkt)->addr = buf->data;
+	net_pkt_lladdr_dst(pkt)->len = sizeof(struct net_can_lladdr);
+	net_pkt_lladdr_dst(pkt)->type = NET_LINK_CANBUS;
+	net_buf_add_be16(buf, get_can_lladdr_dest(frame));
+	net_buf_pull(buf, sizeof(u16_t));
+
+	net_pkt_cursor_init(pkt);
+}
+
+static void net_can_iface_init(struct net_if *iface)
+{
+	struct device *dev = net_if_get_device(iface);
+	struct net_can_context *ctx = dev->driver_data;
+
+	ctx->iface = iface;
+
+	NET_DBG("Init CAN network interface %p dev %p", iface, dev);
+
+	canbus_net_init(iface);
+}
+
+static int net_can_send(struct device *dev, const struct zcan_frame *frame,
+			can_tx_callback_t cb, void *cb_arg, s32_t timeout)
+{
+	struct net_can_context *ctx = dev->driver_data;
+
+	NET_ASSERT(frame->id_type == CAN_EXTENDED_IDENTIFIER);
+	return can_send(ctx->can_dev, frame, timeout, cb, cb_arg);
+}
+
+static void net_can_rcv(struct zcan_frame *frame, void *arg)
```

Comment:
please use _recv instead of _rcv (for consistency with other part of networking code)